integration/ingester_test.go

```diff
@@ -0,0 +1,336 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+// Provenance-includes-location: https://github.com/cortexproject/cortex/blob/master/integration/ingester_sharding_test.go
+// Provenance-includes-license: Apache-2.0
+// Provenance-includes-copyright: The Cortex Authors.
+//go:build requires_docker
+
+package integration
+
+import (
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/grafana/e2e"
+	e2edb "github.com/grafana/e2e/db"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/model/labels"
+	"github.com/prometheus/prometheus/prompb"
+	"github.com/prometheus/prometheus/storage/remote"
+	"github.com/stretchr/testify/require"
+
+	"github.com/grafana/mimir/pkg/util/test"
+
+	"github.com/grafana/mimir/integration/e2emimir"
+)
+
+func TestIngesterMixedFloatHistogramSeries(t *testing.T) {
+	query := "foobar"
+	queryEnd := time.Now().Round(time.Second)
+	queryStart := queryEnd.Add(-1 * time.Hour)
+	queryStep := 10 * time.Minute
+
+	testCases := map[string]struct {
+		inSeries []prompb.TimeSeries
+		expected model.Matrix
+	}{
+		"float series": {
+			inSeries: []prompb.TimeSeries{
+				{
+					Labels: []prompb.Label{
+						{
+							Name:  "__name__",
+							Value: query,
```

Comment:
While the variable is the same, it is confusing to have `query` here (given the `expected` doesn't use `query` too)
```suggestion
							Value: "foobar",
```